File: blog/src/main/kotlin/com/liux/blog/bean/po/Category.kt
package com.liux.blog.bean.po

import com.liux.blog.bean.NoArgsConstructor
import java.io.Serializable
import java.util.*

// 分类数据
@NoArgsConstructor
data class Category(
    // ID
    var id: Int,
    // 名称
    var name: String? = null,
    // 创建时间
    var createTime: Date? = null,

    // 文章数量
    var articleCount: Int? = null,
    // 文章列表
    var articles: List<Article>? = null,
) : Serializable

File: blog/bean/html_vo/category.go
package html_vo

import (
	"blog/bean/po"
)

type CategoryVO struct {
	Name         string
	ArticleCount int
}

func (c *CategoryVO) From(category po.Category) {
	c.Name = category.Name
	c.ArticleCount = len(category.Articles)
}
